src/Npgsql/ThrowHelper.cs

```diff
@@ -44,6 +44,14 @@ internal static void ThrowObjectDisposedException(string objectName, string mess
     internal static void ThrowObjectDisposedException(string objectName, Exception? innerException)
         => throw new ObjectDisposedException(objectName, innerException);
 
+    [DoesNotReturn]
+    internal static void ThrowInvalidCastException(string message)
```

Comment:
See the overload below which accepts the argument as a parameter; the idea is not to have to format the string outside unless the exception actually occurs.